platform/security/sts/security-sts-ldaplogin/src/main/java/ddf/ldap/ldaplogin/SslLdapLoginModule.java

```diff
@@ -477,22 +479,61 @@ protected LDAPConnectionFactory createLdapConnectionFactory(String url, Boolean
         return new LDAPConnectionFactory(host, port, lo);
     }
 
-    /**
-     * Added additional logging to the security LOGGER.
-     */
-    @Override
-    public boolean login() throws LoginException {
-        try {
-            boolean isLoggedIn = doLogin();
-
-            if (!isLoggedIn) {
-                SecurityLogger.audit("Username [" + user + "] failed LDAP authentication.");
+    void validateUsername(String username) throws LoginException {
+        boolean hasBadCharacters = false;
+        for (int i = 0; i < username.length(); i++) {
+            char curChar = username.charAt(i);
+            switch (curChar) {
+            case '\\':
+            case ',':
+            case '+':
+            case '"':
+            case '<':
+            case '>':
+            case ';':
+            case '#':
+                hasBadCharacters = true;
+                break;
             }
-            return isLoggedIn;
+            if (hasBadCharacters) {
+                throw new LoginException("Bad characters present in user name.");
+            }
+        }
+    }
 
-        } catch (Exception le) {
-            throw new LoginException("Username [" + user
-                    + "] could not log in successfuly using LDAP authentication due to an exception");
+    Set<Principal> getPrincipals() {
+        return ImmutableSet.copyOf(principals);
+    }
+
+    protected char[] getDecryptedPassword(String encryptedPassword) {
```

Comment:
Created a method for decrypting the password so that it can be overridden by the test.